src/pyhf/workspace.py

```diff
@@ -3,33 +3,41 @@
 from . import exceptions
 from . import utils
 from .pdf import Model
+from .mixins import _ChannelSummaryMixin
 
 logging.basicConfig()
 log = logging.getLogger(__name__)
 
 
-class Workspace(object):
+class Workspace(_ChannelSummaryMixin, dict):
```

Comment:
is inheriting from `dict` necessary only to make w.get and w[] working?

Reply:
It's both for that and to make `json.dumps(workspace)` possible :)

Reply:
cool!

Reply:
The end goal, if it's not obvious, is that unlike `pyhf.Model` which doesn't currently have a spec.. I want the `pyhf.Workspace` to be a "light" layer on top of the `workspace.json` spec we have. This means that we should do (directly)

```python
spec = pyhf.Workspace(json.load(open('spec.json')))
assert spec['version'] == 1.0.0
```

where it quacks like a python dictionary, but has some extra niceties such as being able to build a model from it, or listing summary information about itself.

Reply:
You might ask, well, why not just provide a `pyhf.Workspace::to_json()` method instead?

- this will break when doing nested `json.dump()` calls

Or some sort of `__json__` that gets called?

- all python json libraries do something different and the one we use doesn't call any magic function and instead expects that we pass in a custom encoder/decoder to the `json.dump()` calls...